rust/feature-flags/src/handler/flags.rs

```diff
@@ -10,29 +10,181 @@ use crate::{
     },
 };
 use axum::extract::State;
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use uuid::Uuid;
 
 use super::{evaluation, types::FeatureFlagEvaluationContext};
 use crate::router;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EvaluationRuntime {
+    All,
+    Client,
+    Server,
+}
+
+impl From<String> for EvaluationRuntime {
+    fn from(s: String) -> Self {
+        EvaluationRuntime::from(s.as_str())
+    }
+}
+
+impl From<&str> for EvaluationRuntime {
+    fn from(s: &str) -> Self {
+        if s.eq_ignore_ascii_case("client") {
+            EvaluationRuntime::Client
+        } else if s.eq_ignore_ascii_case("server") {
+            EvaluationRuntime::Server
+        } else {
+            EvaluationRuntime::All
+        }
+    }
+}
+
+/// Determines the evaluation runtime based on request characteristics.
+/// Uses explicit runtime if provided, otherwise analyzes user-agent, headers, and lib_version
+/// to detect if the request is from a client-side (browser/mobile) or server-side SDK.
+fn detect_evaluation_runtime_from_request(
+    headers: &axum::http::HeaderMap,
+    lib_version: Option<&str>,
+    explicit_runtime: Option<EvaluationRuntime>,
+) -> Option<EvaluationRuntime> {
+    // Use explicitly passed runtime if available
+    if let Some(runtime) = explicit_runtime {
+        return Some(runtime);
+    }
+
+    // Analyze User-Agent header
+    if let Some(user_agent) = headers.get("user-agent").and_then(|v| v.to_str().ok()) {
+        // Browser patterns - these typically indicate client-side execution
+        if user_agent.contains("Mozilla/")
+            || user_agent.contains("Chrome/")
+            || user_agent.contains("Safari/")
+            || user_agent.contains("Firefox/")
+            || user_agent.contains("Edge/")
```

Comment:
I assume IOS, posthog-rn, and Flutter would be receive client flags (see below for why I didn't include Android here yet)

```suggestion
            || user_agent.contains("Edge/")
            || user_agent.starts_with("posthog-react-native/")
            || user_agent.starts_with("posthog-flutter/")
            || user_agent.starts_with("posthog-ios/")
```

1. [Here's where Flutter sets `$sdkName` to `posthog-flutter`](https://github.com/PostHog/posthog-flutter/blob/12aae3c3459ffdd71806c4cbc6beba476e72284b/android/src/main/kotlin/com/posthog/flutter/PosthogFlutterPlugin.kt#L273)
2. [Here's where react-native sets its library id to `posthog-react-native` which gets used to build the user-agent](https://github.com/PostHog/posthog-js/blob/7183893a7f89fbaaab5e65d9e2e6986cf9d95bf2/packages/react-native/src/posthog-rn.ts#L187-L198)